vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -149,6 +142,19 @@ void AdsImpl::InitializeStep4(const Result result) {
   ads_serve_->DownloadCatalog();
 }
 
+void AdsImpl::CleanPostedAsRegistryAfterReboot() {
```

Comment:
Please refactor function name to help with code readability, such as `RemoveAllNotificationsIfNeeded` and decouple business logic so each use case is testable. i.e. stripping business logic for detecting reboot into a function name such as `IsNotificationsOlderThanSystemUptime`.

This will remove the need for the comment as the code will be self-documented and also help with unit testing.

```
void AdsImpl::RemoveAllNotificationsIfNeeded() {
  if (!IsAndroid() ||
      !IsNotificationsOlderThanSystemUptime()) {
    return;
  }

  notifications_->RemoveAll();
}

bool AdsImpl::IsNotificationsOlderThanSystemUptime() {
  auto ads_shown_history = client_->GetAdsShownHistory();
  if (ads_shown_history.empty()) {
    return false;
  }

  uint64_t ad_shown_timestamp_in_seconds = ads_shown_history.front();

  uint64_t boot_timestamp_in_seconds = Time::NowInSeconds() -
      static_cast<uint64_t>(base::SysInfo::Uptime().InSeconds());

  if (ad_shown_timestamp_in_seconds > boot_timestamp_in_seconds) {
    return false;
  }

  return true;
}
```

`IsMobile` function needs refactoring so you can detect Android such as:

```
bool AdsImpl::IsMobile() const {
  if (!IsAndroid() && !IsIOS()) {
    return false;
  }

  return true;
}

bool AdsImpl::IsAndroid() const {
  ClientInfo client_info;
  ads_client_->GetClientInfo(&client_info);

  if (client_info.platform != ANDROID_OS) {
    return false;
  }

  return true;
}

bool AdsImpl::IsIOS() const {
  ClientInfo client_info;
  ads_client_->GetClientInfo(&client_info);

  if (client_info.platform != IOS) {
    return false;
  }

  return true;
}
```

Reply:
1. Renamed to `RemoveAllNotificationsAfterReboot`.
2. We don't need to test detecting system uptime, since it's tested in Chromium code.
3. Used preprocessor to detect target at compile time rather than at run time.

Reply:
3. Can you please use the run time methods as this allows easy testing of such functionality for mobile while developing in desktop

Reply:
The code includes Android specific headers which are not available for desktop build. 
Resolved during compile-time by preprocessor.